app/javascript/react/store/fixedSessionsSelectors.ts

```diff
@@ -1,91 +1,158 @@
 import { createSelector } from "reselect";
-
 import { StatusEnum } from "../types/api";
 import { Session, SessionList } from "../types/sessionType";
 import { RootState } from "./";
+import { FixedSession } from "./fixedSessionsSlice";
+import { FixedSessionsTypes } from "./sessionFiltersSlice";
+
+const selectActiveFixedSessionsState = (state: RootState) =>
+  state.fixedSessions.activeSessions;
+
+const selectDormantFixedSessionsState = (state: RootState) =>
+  state.fixedSessions.dormantSessions;
 
-const selectFixedSessionsState = (state: RootState) => state.fixedSessions;
 const selectFixedSessionsStatusFulfilled = (state: RootState) =>
   state.fixedSessions.status === StatusEnum.Fulfilled;
 
-const selectFixedSessionsPoints = createSelector(
+const selectFixedSessionsState = (state: RootState) => state.fixedSessions;
+
+const selectIsActiveSessionsFetched = createSelector(
+  [selectFixedSessionsState],
+  (fixedSessionsState) => fixedSessionsState.isActiveSessionsFetched
+);
+
+const selectIsDormantSessionsFetched = createSelector(
   [selectFixedSessionsState],
-  (fixedSessionsState): Session[] =>
-    fixedSessionsState.sessions.map(
+  (fixedSessionsState) => fixedSessionsState.isDormantSessionsFetched
+);
+
+const transformSessionData = (sessions: FixedSession[]) =>
+  sessions.map(
+    ({
+      id,
+      title,
+      lastMeasurementValue,
+      startTimeLocal,
+      endTimeLocal,
+      latitude,
+      longitude,
+      streams,
+    }) => {
+      const firstStream = streams[Object.keys(streams)[0]];
+      return {
+        id,
+        title,
+        sensorName: firstStream.sensorName,
+        lastMeasurementValue,
+        startTime: startTimeLocal,
+        endTime: endTimeLocal,
+        averageValue: firstStream.streamDailyAverage,
+        point: {
+          lat: latitude,
+          lng: longitude,
+          streamId: firstStream.id.toString(),
+        },
+        streamId: firstStream.id,
+      };
+    }
+  );
+
+const selectSessionsByType = createSelector(
+  [
+    selectActiveFixedSessionsState,
+    selectDormantFixedSessionsState,
+    (_: RootState, type: FixedSessionsTypes) => type,
+  ],
+  (activeSessions, dormantSessions, type) => {
+    return type === FixedSessionsTypes.ACTIVE
+      ? activeSessions
+      : dormantSessions;
+  }
+);
```

Comment:
you could separate active and dormant session concerns, but as I see further in the code you use them both, so it might not bring any value actually except for slight improvement in performance.